joomla/joomla-cms
# [4.0] [plg_system_httpheaders] Drop X-XSS-Protection and X-Content-Type-Options from the plugin

Pull Request for Issue #25735 && https://scotthelme.co.uk/security-headers-updates/

### Summary of Changes

Drop X-XSS-Protection and X-Content-Type-Options options from the plugin.

#### X-XSS-Protection

> As the most popular platform that still runs an XSS Auditor, Chrome announced plans to drop it in an Intent to Deprecate and Remove: XSS Auditor. There are further details available in this bug but the TLDR is that the XSS Auditor is going. Recent changes included taking it from block mode to filter mode by default and it seems that wasn't enough to save it from the false positives. Edge removed their XSS Filter last year so with the removal from Chrome there is now no browser that will have a native XSS protection. It's time for the header to go! With that said, and using my site as an example, you can see that the X-Xss-Protection header is no longer required for an A+ grade on Security Headers.

https://scotthelme.co.uk/security-headers-updates/

According to this the header is dying and was removed in Edge already and is now going to be remove in Chrome too. It makes no sense to ship Joomla 4 with that old thing that when we anyway only support the latest version of browsers. Based on that text it also seams that chrome in the past already defaulted to block mode now filter mode so older versions are also "protected" as long as you are using them. The best way to eliminate XSS is using CSP which we make much easier to do in Joomla 4 anyway.

#### X-Content-Type-Options

This header is now enforced by the default htaccess and web.config and seams to only confuse people as the switch has no affect at all anymore https://github.com/joomla/joomla-cms/issues/25735

### Testing Instructions

Apply this patch, make sure both headers are not possible to configure now and not set anymore.

### Expected result

both headers are not possible to configure now and not set anymore.

### Actual result

both headers are possible to configure now and get set as of today.

### Documentation Changes Required

Yes.

### Additional Info

I'm happy for all your input on this and this is a suggestion to deal with that two issues.

cc @SniperSister  :)
